[MrTwit99] iP

MrTwit99 opened the pull request:
## Duke
>“Ideas are easy. Implementation is hard.”  – Guy Kawasaki, Alltop Co-Founder and Entrepreneur [(source)](https://www.forbes.com/sites/allbusiness/2014/02/10/50-inspirational-quotes-for-startups-and-entrepreneurs/?sh=244f8cd56ef0)
>
Do you **KNOW** what else is **HARD**? 
Remembering all the ideas that you have and finding a place to **STORE** them and **LOCATE** them! 
**FEAR NOT** as ***Duke*** is here to resolve **ALL** these **ISSUES** with some simple clicks!

![image](https://user-images.githubusercontent.com/97277438/216394418-87e9aaab-571f-4e3e-b626-82f08b4b9c4f.png)
[(source)](https://www.google.com/url?sa=i&url=https%3A%2F%2Fconsequence.net%2F2021%2F01%2Frowan-atkinson-mr-bean-stressful-exhausting%2F&psig=AOvVaw10DpW23S9hQ-TyUORAQGRU&ust=1675444257951000&source=images&cd=vfe&ved=0CBAQjRxqFwoTCJjloLCq9_wCFQAAAAAdAAAAABAE)

DukePro frees your mind of having to remember things you need to do. It's,

+ text-based
+ easy to pick up
+ ~~FAST~~ SUPER FAST to use

DukePro currently supports the following functions:

- [X] Managing tasks
- [X] Managing deadlines
- [X] Managing event details
- [X] Search / Filter for any task / deadline / event
- [X] Reminders
- [X] !help

## Instructions for Installation
All you need to do is,

1. Visit the website [here](https://github.com/MrTwit99/ip.git).
2. Click the section **Releases** and download the latest version's `duke.jar`.
3. Move the downloaded `duke.jar` into a new folder on your computer.
4. Double click `duke.jar` to run the application
    OR 
    Alternatively run the following command in your command prompt window.

    ```ruby
    java -jar duke.jar
    ```
5. Add your tasks.
6. Let it track and manage your tasks for you! :sunglasses: :thinking: :wink:

##
View a snippet of the `main` method here:
```java
public static void main(String[] args) throws IOException {
    new Duke("data/storage.txt", "data").run();
}
```

Commit:
Add Gradle support

Commit:
Updated Duke with Level-1: Greet, Echo, Exit

Commit:
Updated Level-1 with improvements to it!

Commit:
Updated Level-1 with its tag

Commit:
Updated Duke with Level-2: Add, List

Commit:
Updated Duke with Level-3: Mark as Done

Commit:
Updated Duke with Level-4: ToDos, Events, Deadlines and added classes Event, ToDos, Task and Deadline

Commit:
Updated input.txt and EXPECTED.TXT with the required Automated Text UI Testing with testcase from Level-4 example!

Commit:
Updated Duke with Level-6: Delete, not yet done with Level-5: Handle Errors!

Commit:
Updated Duke with Level-5: Handle Errors and added DukeException.java for exceptions unique to Duke

Commit:
Updated functions "mark", "unmark" and "delete" to handle situations where the task the user is trying to select is OUT OF RANGE!

Commit:
Updated Duke.java (amended some texts and shorten some code) and Task.java (amended some texts)

Commit:
Updated Testcases to use for testing in EXPECTED.TXT and input.txt
Updated Duke.java and DukeException.java (shortened Duke.java code with less repetition of calling try catch statement for the same thing)

Commit:
Fixed Duke.java switch case's text indentation.

Commit:
Updated imports in Duke.java and DukeException.java such that they are listed explicitly.

Example import java.util.*; > import java.util.ArrayList;

Commit:
Created a new folder "data" to contain "storage.txt".
Storage.txt is used to store all tasks created by the user. This can normally be viewed via the command "list".
Updated Duke.java and DukeException.java to check if the folder "data" exists and if "storage.txt" exists within data.

Commit:
Fix DukeException.java's method definitions form by following the Java coding standards

Commit:
Fix Duke.java's boolean variable namings and minor issues with regards to Java coding standards

Commit:
Fix some minor bugs with Level-7: Save, left with writing to file and overriding to the file

Commit:
Update the following classes with Level-7 requirements: Deadline.java, Duke.java, Event.java, Task.java, ToDos.java
Update storage.txt

Add functions for Level-7: Writing to File, Retrieving from File, Overwriting of file

Fix some minor bugs and errors pertaining to Java Coding Standards

Commit:
Update the following classes with Date and Time as per Level-8 requirements but without the ability to print deadlines/events occurring on a specific date: Deadline.java, Duke.java, Event.java

Commit:
Merge branch 'branch-Level-7'

Commit:
Update tasks stored in storage.txt

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

Commit:
Fix some minor issues with Duke.java

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Create packages: classes and exceptions to store the respective java files

Move Deadline.java, Duke.java, Event.java, Task.java, ToDos.java to package classes
Move DukeException.java to package exceptions

Add FolderNotFoundException.java, IncorrectNoOfArgumentException.java, InvalidCommandException.java to package exceptions

Commit:
Add DeadlineTest.java, EventTest.java, ToDosTest.java to test folder as Junit tests!

Commit:
Create the JAR file but there's some issues with its file reading and writing ability, mainly file could not be located

Commit:
Fix Duke.java and DukeException.java to handle and resolve when one cannot locate a specific directory / file

Commit:
Update JAR file with the new working copy after fixing issues with missing folder and file

mmaimer33 commented on data/storage.txt:
```diff
@@ -0,0 +1,6 @@
+[T][ ] borrow book
+[D][ ] Assignment 1 /by 2023-12-12 
+[D][X] Assignment 1 /by 2023-12-12 12:40
+[E][ ] Birthday /from 2023-05-29 12:30 /to 2024-04-23 13:00
+[T][X] sleep
+[T][ ] eat
```
May be better to add this to the .gitignore

MrTwit99 replied:
Alright, thanks for the advice! I will incorporate it into future updates!

mmaimer33 commented on src/main/java/classes/Duke.java:
```diff
@@ -0,0 +1,424 @@
+
+public class Duke {
+
+    private static ArrayList<String> getFileContents(String filePath, String folderPath) throws IOException,
```
Would it be better to put this in a separate class with other storage functionality?

MrTwit99 replied:
I have yet to update with "A-MoreOOP" section, do stay tune to its release.

mmaimer33 commented on src/main/java/classes/Duke.java:
```diff
@@ -0,0 +1,424 @@
+            System.out.println("An unexpected error has occurred: " + e.getMessage());
+        } finally {
+            FileWriter fw = new FileWriter("data/storage.txt");
+            if (fileTasks.size() != 0) {
+                for (int i = 0; i < fileTasks.size(); i++) {
+                    if ( (fileTasks.get(i).equals(oldText)) && (i == oldTextIndex) ) {
+                        if (newText.equals("")) {
+                            continue;
+                        }
+                        fw.write(newText + "\n");
+                        continue;
+                    }
+                    fw.write(fileTasks.get(i) + "\n");
```
Could this perhaps be simplified?

MrTwit99 replied:
I have yet to update with "A-MoreOOP" section, do stay tune to its release.

mmaimer33 commented on src/main/java/classes/Duke.java:
```diff
@@ -0,0 +1,424 @@
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        PrintWriter pw = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
+        StringBuilder sb = new StringBuilder();
+        boolean isFileData = true;  // whether to retrieve tasks from the saved file
+        ArrayList<String> fileTasks = new ArrayList<>();
+        ArrayList<Task> taskList = new ArrayList<>();
+        String text;
```
Would recommend breaking this up into more readable fragments.

MrTwit99 replied:
I have yet to update with "A-MoreOOP" section, do stay tune to its release.

mmaimer33 commented on src/main/java/classes/Duke.java:
```diff
@@ -0,0 +1,424 @@
+                break;
+            default:    // add new Task into the Reminder list
+                Task newTask = null;
+                switch (tempCmd) {
```
This seems slightly repetitive; might you consider re-using previously-written abstractions for the default case?

mmaimer33 commented on src/main/java/classes/Event.java:
```diff
@@ -0,0 +1,62 @@
+    public Event(String taskInfo, String stringStartDate, String stringEndDate, String stringStartTime
+            , String stringEndTime) {
+        super(taskInfo);
+        this.oldStartDate = stringStartDate;
+        this.oldEndDate = stringEndDate;
+        this.oldStartTime = " " + stringStartTime;
+        this.oldEndTime = " " + stringEndTime;
+        this.startDate = LocalDate.parse(stringStartDate);
+        this.newStartDate = DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG)
+                .format(startDate);
+        this.endDate = LocalDate.parse(stringEndDate);
+        this.newEndDate = DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG)
+                .format(endDate);
```
The constructor for this and other Task classes is quite difficult to understand, is there any way of making it clearer?

MrTwit99 replied:
Alright, thanks for the advice! I will review through this section and release an improved version!

mmaimer33 commented on src/test/java/classes/EventTest.java:
```diff
@@ -0,0 +1,85 @@
+    Task tempTask4 = new Event("Birthday", "2023-02-05", "2023-02-05", "", "");
+
+    @Test
+    public void getTaskInfoStatus() {
```
Would recommend breaking this into several tests for readability.

MrTwit99 replied:
Could you perhaps clarify more on which part of the code that could be broken down into several tests?

mmaimer33 reviewed (commented):
The program is complete and functional, but there are some elements that could be improved. Adding the "A-MoreOOP" section of week 3 might make the code more organised and easier to go through.

Commit:
Add classes: Parser.java, Storage.java, TaskList.java, Ui.java, DukeTest.java to incorporate OOP onto Duke.java

Edit DukeException.java to fix some inputs to methods validate(). TAKE NOTE of the arguments required by the NEW validate().

*Additional comments: storage.txt is no longer being revision controlled and validate() takes in DIFFERENT arguments now!

huixuant commented on src/main/java/classes/Parser.java:
```diff
@@ -0,0 +1,164 @@
+public class Parser {
+
+    protected static ArrayList<String> parse(String taskInfo) throws IncorrectNoOfArgumentException {
+        ArrayList<String> parseInfo = new ArrayList<>();    // stores in the format "command" followed by "arguments"
```
Perhaps a more intuitive variable name like `arguments` could be used here? `parseInfo` sounds like a verb/action instead of a noun, and the coding standard also recommends variables for such collections to be in plural form.

MrTwit99 replied:
Alright, thanks for the advice! I'm not sure if the new name fits better but I called it "parseInfoList" ahahaha! 
Do let me know if you have any recommended names!

huixuant commented on src/main/java/classes/Storage.java:
```diff
@@ -0,0 +1,119 @@
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+public class Storage {
```
Perhaps it would be good to include JavaDoc header comments for this, as well as all other public classes and methods? This would help users better understand what each class and method is supposed to do.

MrTwit99 replied:
Sorry for any inconvenience caused as you were looking through. 
An updated version with JavaDoc headers incorporated has been pushed! Do take a look at them!

*Note: Classes under the folder "test" has yet gotten their JavaDocs as the test classes are still work in progress.

huixuant commented on src/main/java/classes/ToDos.java:
```diff
@@ -0,0 +1,16 @@
+package classes;
+
+public class ToDos extends Task {
```
Though not explicitly mentioned in the coding standard, perhaps this class should be named `ToDo` instead of `ToDos` (i.e. without the s)? All of the examples for class name given in the standard are singular nouns, and this change would also make the naming of this class consistent with the others in this project (e.g. `Deadline`, `Event`).

MrTwit99 replied:
I agree with you on this. Could I just check with you is the term "ToDo" some sort of reserved word or anything? 
I noticed it highlighted in yellow and everything else aft that turned yellow when i mistyped "ToDos" as "ToDo" when commenting. ( // ...)

huixuant commented on src/main/java/exceptions/DukeException.java:
```diff
@@ -0,0 +1,62 @@
+
+public class DukeException {
+    private static int expectedArgs = 1;
+    public static void validate(boolean secondaryCheck, String command,
```
Perhaps the parameter `secondaryCheck` can be renamed to sound like a boolean (e.g. isXXX, hasXXX)? This is in line with what the coding standard recommends.

MrTwit99 replied:
Apologies for any inconvenience caused as you were looking through.
The previous version was not revised according to Java Coding Standards. 
A new updated version, following the Java Coding Standards, has been released. Do take a look!

huixuant commented on src/main/java/classes/Duke.java:
```diff
@@ -0,0 +1,77 @@
+                continue;
+            }
+            if (commandInfo.size() != 0) {
+                switch (commandInfo.get(0)) {
```
Switch statement is well-formatted according to the coding standard (i.e. no indentation for case clauses). Good job! 😄

huixuant reviewed (commented):
Noted a few coding standard violations, but mainly minor ones that require just a quick fix.

Commit:
Add JavaDoc to the following java files: Storage.java, Task.java, Deadline.java, Duke.java, Event.java, Parser.java, TaskList.java, ToDos.java, Ui.java, DukeException.java, FolderNotFoundException.java, IncorrectNoOfArgumentException.java, InvalidCommandException.java

*Notes: JavaDoc has yet to be added into the files under the test directory.

Commit:
Fix some minor JavaDoc issues with InvalidCommandException.java

Commit:
Amend the following classes code as per the Java Coding Standard: Duke.java, Event.java, Parser.java, Storage.java, Task.java, TaskList.java, DukeException.java

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Add a new function for Duke: find(String str).

*Note: This function is capable of finding all tasks that has the string in its task information and print them out in an ascending order list.

Commit:
Fix a grammar error in TaskList

Commit:
Merge branch 'branch-Level-9'

Commit:
Fix find() in TaskList.java to be able to look for individual words in a given substring instead of a whole string to improve search results.

*Note: Took inspiration from fellow GitHub user ("huixuant") during PR and incorporated this to improve the searching capabilities of find() in TaskList.java.

Commit:
Change the directory name of directory "classes" to "duke" to give a better name for the following classes:
Deadline.java, Duke.java, Event.java, Parser.java, Storage.java, Task.java, TaskList.java, ToDos.java, Ui.java

Commit:
Fix and updated the test files so that they can be used.

Commit:
Add new test file "storageTest(delete before each test).txt" to be ignored in .gitignore

Commit:
Add new file to ignore in .gitignore

Commit:
Change file name "ToDosTest.java" and "ToDos.java" to "ToDoTest.java" and "ToDo.java" respectively and updated TaskList.java code of ToDo

Commit:
Change file "ToDosTest.java" to "ToDoTest.java"

Commit:
Fix the issues with regards to A-CheckStyle guidelines after adding in the Checkstyle files into Duke.

chuakid commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,109 @@
+                    try {
+                        ui.printCommand(tasks.addTask(commandInfoList, this.storage));
+                    } catch (DateTimeParseException e) {
+                        System.out.println("Invalid inputs!\n");
+                        System.out.println("Please enter your date & time in the format: YYYY-MM-DD HH:MM \n");
+                        System.out.println("Please also ensure they are valid values!\n");
```
This part here could use the ui component to print too, to maintain encapsulation and SLAP.

chuakid commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,109 @@
+                    try {
+                        ui.printCommand(tasks.addTask(commandInfoList, this.storage));
+                    } catch (DateTimeParseException e) {
+                        System.out.println("Invalid inputs!\n");
+                        System.out.println("Please enter your date & time in the format: YYYY-MM-DD HH:MM \n");
+                        System.out.println("Please also ensure they are valid values!\n");
```
This part here could use the ui component to print too, to maintain encapsulation and SLAP.

MrTwit99 replied:
Alright, thanks for the advice! I would incorporate this in future commits!

chuakid commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,109 @@
+                System.out.println(e);
+                continue;
+            }
+            if (commandInfoList.size() != 0) {
+                switch (commandInfoList.get(0)) {
+                case "bye":
+                    isExit = true;
+                    ui.showFarewellMessage();
+                    break;
+                case "list":
+                    ui.printCommand(tasks.list());
+                    break;
+                case "find":
+                    ui.printCommand(tasks.find(commandInfoList.get(1)));
+                    break;
+                case "mark":
+                    ui.printCommand(tasks.markTask(commandInfoList, this.storage));
+                    break;
+                case "unmark":
+                    ui.printCommand(tasks.unmarkTask(commandInfoList, this.storage));
+                    break;
+                case "delete":
+                    ui.printCommand(tasks.deleteTask(commandInfoList, this.storage));
+                    break;
+                case "error":
+                    break;
+                default:
+                    try {
+                        ui.printCommand(tasks.addTask(commandInfoList, this.storage));
+                    } catch (DateTimeParseException e) {
+                        System.out.println("Invalid inputs!\n");
+                        System.out.println("Please enter your date & time in the format: YYYY-MM-DD HH:MM \n");
+                        System.out.println("Please also ensure they are valid values!\n");
```
This is a very long and deeply nested function. Maybe you could consider abstracting the commands into different classes?

MrTwit99 replied:
I agree with you on the creation of another class possibly command class with subclasses dealing with the respective command. This would probably be an area I would work on and release for future commits!

chuakid commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,185 @@
+     * @throws IncorrectNoOfArgumentException When there is insufficient arguments provided by the user.
+     * @see ArrayList
+     */
+    protected static ArrayList<String> parse(String taskInfo) throws IncorrectNoOfArgumentException {
```
Very long and complicated function. Maybe you could abstract the parsing of separate parts of the string into individual functions?

MrTwit99 replied:
hmmm I agree with you on that as well. However, I would probably not be fixing this issue anytime soon due to the time constraint. I would look into rectifying this should time permits.

chuakid commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,170 @@
+            Files.createFile(tempFilePath);
+            System.out.println(e);
+        } catch (FileNotFoundException e) {
+            sb.append("    ____________________________________________________________\n")
+                    .append("    File 'storage' cannot be found in the folder 'data'.\n")
+                    .append("    A new file 'storage' has been created for you under the folder 'data'\n")
+                    .append("    for storing the tasks!\n")
+                    .append("    ____________________________________________________________\n");
```
I like this! Very easy on the eyes.

chuakid reviewed (commented):
Good work with some minor improvements on code legibility to be made!

Commit:
Amend Duke.java and Ui.java to encapsulate the details of printing DateTimeFormat error to the user.

Commit:
Edit JavaDocs info for printInvalidDateTime().

Commit:
Add classes: Command, AddCommand, DeleteCommand, FindCommand, ListCommand, MarkCommand, UnmarkCommand to remove the operations from taskList.

Commit:
Merge branch 'branch-Command'

Commit:
Edit TaskList.java and removed a few redundant statements that are not used.

Commit:
Fix TaskList.java by removing one unnecessary code line.

Commit:
Create GUI for Duke.

Update following files to incorporate JavaFX: build.gradle, Duke.java, Parser.java, Ui.java, AddCommand.java, Command.java, DeleteCommand.java, FindCommand.java, ListCommand.java, MarkCommand.java, UnmarkCommand.java.

Add following classes related to JavaFX: DialogBox.java, Launcher.java, Main.java, MainWindow.java, ByeCommand.java.
Add following .fxml classes for JavaFX: DialogBox.fxml, MainWindow.fxml.
Add following images for JavaFX: Shiro.jpg, CrayonShinChan.png

Commit:
Merge branch 'branch-Level-10'

Commit:
Add image_sources.txt

File "image_sources.txt" contains source links to the images used to
portray the user and duke. This is done to give credits to the
original owner who owned the images.

Commit:
Add ip.jar file from ip_jar3 directory to .gitignore

This is prevent committing ip.jar file to the repository.
The file ip.jar will instead be uploaded as a release.

Commit:
Add ip.jar from ip_jar3 directory to .gitignore

To ignore and avoid committing ip.jar file.

Commit:
Edit build.grade to amend property mainClassName

Amend mainClassName content from seedu.duke.Duke to
seedu.duke.Launcher.

Commit:
Edit build.gradle file

To fix the issue of JAR file not opening the GUI when ran.

Commit:
Edit java files: Deadline.java, Duke.java, Event.java, Parser.java,
Storage.java, AddCommand.java, FindCommand.java

AddCommand.java: remove unnecessary line
Deadline.java & Event.java: Fix constructor portion to align with
coding standards
Rest of files: Add the use of assertions

Commit:
Edit the following classes as per the Code Quality standards learnt
in class:

- Duke.java
- Parser.java
- Storage.java
- TaskList.java
- AddCommand.java
- DeleteCommand.java
- FindCommand.java
- MarkCommand.java
- UnmarkCommand.java

MrTwit99 committed:
Merge pull request #1 from MrTwit99/branch-A-Assertions

Add assertions to code and fix some coding standards issue

Commit:
Merge branch 'master' of https://github.com/MrTwit99/ip

* 'master' of https://github.com/MrTwit99/ip:
  Edit java files: Deadline.java, Duke.java, Event.java, Parser.java, Storage.java, AddCommand.java, FindCommand.java

Commit:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/packages/AddCommand.java

Commit:
Fix some minor bugs with assertions used in Duke.java, Storage.java
and FindCommand.java

Commit:
Fix line borders length in the following files:
- Ui.java
- AddCommand.java
- DeleteCommand.java
- FindCommand.java
- ListCommand.java
- MarkCommand.java
- UnmarkCommand.java
- DukeException.java

Commit:
Add ReminderCommand.java to create reminder function
Edit the following classes to create the reminder function:
- Deadline.java
- Duke.java
- Event.java
- Parser.java
- Task.java
- ToDo.java
- FindCommand.java

Notes:
Point 1:
The newly implemented reminder function via ReminderCommand tracks
all Deadline and Event tasks along with their completion status.
Tasks are being checked against the system date and time.
Therefore, ENSURE the system is SET to the CORRECT date and time.

Point 2:
ToDo tasks are also tracked but via their completion status and not
date time since they are not allocated with any!
It is ASSUMED that all ToDo tasks are to be done ASAP!

Commit:
Fix a minor error found during implementation of reminder function.

MrTwit99 committed:
.github/workflows/gradle.yml

MrTwit99 committed:
Delete gradle.yml

MrTwit99 committed:
Create gradle.yml

Commit:
Merge branch 'master' of https://github.com/MrTwit99/ip

* 'master' of https://github.com/MrTwit99/ip:
  Create gradle.yml
  Delete gradle.yml
  .github/workflows/gradle.yml

praveenkrishna0512 commented on src/main/java/duke/packages/FindCommand.java:
```diff
@@ -0,0 +1,68 @@
+            }
+            break;
+        default: // when string is made up of multiple words
+            for (String word: words) {
+                for (int i = 0; i < taskList.size(); i++) {
+                    if (taskList.get(i).getTaskInfo().toLowerCase().contains(word)) {
+                        sb.append("    ").append(taskIndex++).append(".").append(taskList.get(i)
+                                .getTaskInfoStatus()).append("\n");
+                    }
+                }
+            }
```
Perhaps this could be abstracted into its own method!

Commit:
Rename package "packages" to "commands"

Edit eventParser() in Parser.java

Add eventStartParser() and eventEndParser() to Parser.java to aid
eventParser() in parsing the event startDate, startTime, endDate and
endTime.

Commit:
Add ip_jar2/duke.jar to .gitignore

MrTwit99 committed:
Delete artifacts/ip_jar2 directory

Commit:
Fix EventTest.java to apply SLAP

Add JavaDocs to EventTest.java, ToDoTest.java, DeadlineTest.java

Remove DukeTest.java

Commit:
Add Ui.test to JUnit Tests

Edit DeadlineTest.java, EventTest.java, ToDoTest.java

Commit:
Add ParserTest.java as JUnit test to test Parser.java

Edit Parser.java to fix a situation where incorrect number of
arguments given but not detected

Edit DeadlineTest.java and UiTest.java to fix minor text error

Commit:
Edit DukeException.java to fix some minor text issue

Edit DialogBox.java, Main.java, MainWindow.java, image_sources.txt,
DialogBox.fxml, MainWindow.fxml, Ui.png, doraemon.jpg, nabita2.jpg
to implement A-BetterGui

Commit:
Edit MainWindow.java, ParserTest.java

Rename doraemon.jpg to doraemon2.jpg

Commit:
Edit MainWindow.fxml to make button, textfield and anchorpane to be
resizable to Window.

However, some small issue with the text in textfield and button
being shifted whenever resized.

Commit:
Add isAvailable() in Parser.java to check if the command is
available and supported by Duke

Commit:
Edit Duke.java to fix a situation where the application shows
"The handle is invalid" error when the "bye" command is called

Commit:
Add HelpCommand.java for the help command

Edit Duke.java, Parser.java, Ui.java to incorporate help command

Edit Ui.java to fix some minor documentation error

MrTwit99 committed:
Add DukeBootupPhoto.jpg to use for User Guide

MrTwit99 committed:
Add folder for storing images used in User Guide

MrTwit99 committed:
Delete DukeBootupPhoto.jpg

MrTwit99 committed:
Add files via upload

Commit:
Add User Guide to README.md file

*Note: Need to include Command Summary to complete User Guide

Commit:
Edit README.md to fix a small issue

Commit:
Edit README.md to implement anchor link to different sections of
the User Guide

Commit:
Update README.md with Command Summary

*Note:
Completed the User Guide for iP

Commit:
Edit some minor display issue for README.md

Commit:
Edit README.md to fix Table Creation issue for Command Summary

MrTwit99 committed:
Delete test.txt

MrTwit99 committed:
Delete README.md
